File: iOS/SLC-Lovers/Views/Home/HomeView.swift
import SwiftUI

/// 首页
struct HomeView: View {
    @EnvironmentObject var appState: AppState
    @State private var now = Date()

    private let timer = Timer.publish(every: 60, on: .main, in: .common).autoconnect()

    var body: some View {
        NavigationView {
            ScrollView {
                VStack(spacing: SLCSpace.lg) {
                    // 顶部问候
                    HeaderSection(userName: appState.currentUser?.name ?? "我")

                    // 在一起大数字
                    TogetherCounter(startDate: appState.pairing?.startDate ?? Date())

                    // 每日寄语
                    if let quote = appState.todayQuote {
                        QuoteCard(quote: quote)
                    }

                    // 下一个纪念日
                    if let next = nextAnniversary() {
                        NextAnniversaryCard(anniversary: next)
                    }

                    // 待办预览
                    TodoPreviewSection()

                    // 愿望清单预览
                    BucketPreviewSection()

                    // 最近聊天
                    RecentChatSection()

                    Spacer(minLength: SLCSpace.xxl)
                }
                .padding(.vertical, SLCSpace.md)
            }
            .background(SLCColor.cream)
            .navigationBarHidden(true)
            .onReceive(timer) { _ in
                now = Date()
            }
            .onAppear {
                appState.refreshDailyQuote()
            }
        }
    }

    private func nextAnniversary() -> Anniversary? {
        appState.anniversaries
            .filter { !$0.title.contains("在一起") }
            .min(by: { $0.daysUntilNext() < $1.daysUntilNext() })
    }
}

// MARK: - 顶部
private struct HeaderSection: View {
    let userName: String

    var body: some View {
        HStack {
            VStack(alignment: .leading, spacing: 4) {
                Text(greeting())
                    .font(SLCFont.body(13))
                    .foregroundColor(SLCColor.textSecondary)
                Text(userName)
                    .font(SLCFont.title(SLCFontSize.titleLarge, weight: .semibold))
                    .foregroundColor(SLCColor.textPrimary)
            }
            Spacer()
            NavigationLink(destination: SettingsView()) {
                Image(systemName: "person.crop.circle.fill")
                    .font(.system(size: 36))
                    .foregroundColor(SLCColor.him)
            }
        }
        .padding(.horizontal, SLCSpace.lg)
    }

    private func greeting() -> String {
        let hour = Calendar.current.component(.hour, from: Date())
        switch hour {
        case 5..<11: return "早安，"
        case 11..<14: return "中午好，"
        case 14..<18: return "下午好，"
        case 18..<22: return "晚上好，"
        default: return "夜深了，"
        }
    }
}

// MARK: - 在一起大数字
private struct TogetherCounter: View {
    let startDate: Date

    var body: some View {
        VStack(spacing: SLCSpace.sm) {
            Text("在一起")
                .font(SLCFont.body(14, weight: .medium))
                .foregroundColor(SLCColor.textSecondary)
                .tracking(4)

            HStack(alignment: .lastTextBaseline, spacing: 4) {
                Text(DateUtils.togetherDays(since: startDate))
                    .font(.system(size: 96, weight: .ultraLight, design: .serif))
                    .foregroundColor(SLCColor.him)
                    .contentTransition(.numericText())
                Text("天")
                    .font(SLCFont.title(28, weight: .light))
                    .foregroundColor(SLCColor.him)
            }

            Text(DateUtils.fullChinese(startDate))
                .font(SLCFont.caption(SLCFontSize.bodySmall))
                .foregroundColor(SLCColor.textSecondary)
        }
        .frame(maxWidth: .infinity)
        .padding(.vertical, SLCSpace.xl)
        .background(
            LinearGradient(
                colors: [SLCColor.creamLight, SLCColor.cream],
                startPoint: .top,
                endPoint: .bottom
            )
        )
        .clipShape(RoundedRectangle(cornerRadius: SLCRadius.xl))
        .padding(.horizontal, SLCSpace.lg)
        .slcShadow(.soft)
    }
}

// MARK: - 每日寄语
private struct QuoteCard: View {
    let quote: Quote

    var body: some View {
        SLCCard {
            HStack(alignment: .top, spacing: SLCSpace.md) {
                Image(systemName: "quote.opening")
                    .font(.system(size: 24))
                    .foregroundColor(SLCColor.herDeep.opacity(0.5))
                VStack(alignment: .leading, spacing: 4) {
                    Text(quote.content)
                        .font(SLCFont.body(SLCFontSize.bodyLarge, weight: .regular))
                        .foregroundColor(SLCColor.textPrimary)
                        .lineSpacing(4)
                }
            }
        }
        .padding(.horizontal, SLCSpace.lg)
    }
}

// MARK: - 下一个纪念日
private struct NextAnniversaryCard: View {
    let anniversary: Anniversary

    var body: some View {
        SLCCard {
            HStack(spacing: SLCSpace.md) {
                Text(anniversary.icon)
                    .font(.system(size: 44))
                VStack(alignment: .leading, spacing: 4) {
                    Text(anniversary.title)
                        .font(SLCFont.title(SLCFontSize.titleSmall, weight: .semibold))
                        .foregroundColor(SLCColor.textPrimary)
                    HStack(spacing: 6) {
                        Text("还有")
                            .font(SLCFont.caption(SLCFontSize.bodySmall))
                            .foregroundColor(SLCColor.textSecondary)
                        Text("\(anniversary.daysUntilNext())")
                            .font(.system(size: 20, weight: .medium, design: .serif))
                            .foregroundColor(SLCColor.herDeep)
                        Text("天")
                            .font(SLCFont.caption(SLCFontSize.bodySmall))
                            .foregroundColor(SLCColor.textSecondary)
                    }
                }
                Spacer()
                NavigationLink(destination: AnniversaryListView()) {
                    Image(systemName: "chevron.right")
                        .font(.system(size: 14, weight: .semibold))
                        .foregroundColor(SLCColor.textTertiary)
                }
            }
        }
        .padding(.horizontal, SLCSpace.lg)
    }
}

// MARK: - 待办预览
private struct TodoPreviewSection: View {
    @EnvironmentObject var appState: AppState

    var pendingTodos: [TodoItem] {
        appState.todos.filter { !$0.isCompleted }.prefix(3).map { $0 }
    }

    var body: some View {
        VStack(alignment: .leading, spacing: SLCSpace.sm) {
            SLCSectionHeader(title: "待办", action: "全部") {
                // TODO: 跳转到全部待办
            }
            if pendingTodos.isEmpty {
                Text("暂无待办，去记下今天想做的事吧")
                    .font(SLCFont.body(SLCFontSize.bodyMedium))
                    .foregroundColor(SLCColor.textSecondary)
                    .frame(maxWidth: .infinity, alignment: .center)
                    .padding(SLCSpace.lg)
                    .background(SLCColor.creamLight)
                    .clipShape(RoundedRectangle(cornerRadius: SLCRadius.lg))
                    .padding(.horizontal, SLCSpace.md)
            } else {
                VStack(spacing: SLCSpace.sm) {
                    ForEach(pendingTodos) { todo in
                        TodoRow(todo: todo) {
                            appState.toggleTodo(todo)
                        }
                    }
                }
                .padding(.horizontal, SLCSpace.md)
            }
        }
    }
}

private struct TodoRow: View {
    let todo: TodoItem
    let onToggle: () -> Void

    var body: some View {
        HStack(spacing: SLCSpace.md) {
            Button(action: {
                Haptics.selection()
                onToggle()
            }) {
                Image(systemName: "circle")
                    .font(.system(size: 22))
                    .foregroundColor(SLCColor.textTertiary)
            }
            VStack(alignment: .leading, spacing: 2) {
                Text(todo.title)
                    .font(SLCFont.body(SLCFontSize.bodyLarge))
                    .foregroundColor(SLCColor.textPrimary)
                if let date = todo.dueDate {
                    Text(DateUtils.monthDay(date))
                        .font(SLCFont.caption(SLCFontSize.bodySmall))
                        .foregroundColor(SLCColor.textSecondary)
                }
            }
            Spacer()
            if let assignee = todo.assignee {
                SLCPersonBadge(role: assignee, compact: true)
            }
        }
        .padding(SLCSpace.md)
        .background(SLCColor.creamLight)
        .clipShape(RoundedRectangle(cornerRadius: SLCRadius.md))
    }
}

// MARK: - 愿望清单预览
private struct BucketPreviewSection: View {
    @EnvironmentObject var appState: AppState

    var activeBuckets: [BucketItem] {
        appState.bucketItems.filter { !$0.isAchieved }
    }

    var body: some View {
        VStack(alignment: .leading, spacing: SLCSpace.sm) {
            SLCSectionHeader(title: "愿望清单")
            ScrollView(.horizontal, showsIndicators: false) {
                HStack(spacing: SLCSpace.sm) {
                    ForEach(activeBuckets.prefix(5)) { bucket in
                        NavigationLink(destination: CapsuleView()) {
                            BucketCard(bucket: bucket)
                        }
                        .buttonStyle(.plain)
                    }
                }
                .padding(.horizontal, SLCSpace.md)
            }
        }
    }
}

private struct BucketCard: View {
    let bucket: BucketItem

    var body: some View {
        VStack(alignment: .leading, spacing: SLCSpace.sm) {
            Text(bucket.title)
                .font(SLCFont.title(SLCFontSize.bodyLarge, weight: .medium))
                .foregroundColor(SLCColor.textPrimary)
                .lineLimit(2)
                .multilineTextAlignment(.leading)
                .frame(width: 160, alignment: .leading)
            if let desc = bucket.description {
                Text(desc)
                    .font(SLCFont.caption(SLCFontSize.bodySmall))
                    .foregroundColor(SLCColor.textSecondary)
                    .lineLimit(1)
            }
        }
        .padding(SLCSpace.md)
        .frame(width: 180, height: 100, alignment: .topLeading)
        .background(SLCColor.creamLight)
        .clipShape(RoundedRectangle(cornerRadius: SLCRadius.lg))
    }
}

// MARK: - 最近聊天预览
private struct RecentChatSection: View {
    @EnvironmentObject var appState: AppState

    var lastMessage: Message? {
        appState.messages.last
    }

    var body: some View {
        if let last = lastMessage {
            VStack(alignment: .leading, spacing: SLCSpace.sm) {
                SLCSectionHeader(title: "最近的消息")
                NavigationLink(destination: ChatView()) {
                    SLCCard {
                        HStack(spacing: SLCSpace.md) {
                            Image(systemName: "bubble.left.fill")
                                .foregroundColor(SLCColor.herDeep)
                                .font(.system(size: 22))
                            VStack(alignment: .leading, spacing: 2) {
                                Text(last.content)
                                    .font(SLCFont.body(SLCFontSize.bodyLarge))
                                    .foregroundColor(SLCColor.textPrimary)
                                    .lineLimit(1)
                                Text(DateUtils.friendlyRelative(from: last.createdAt))
                                    .font(SLCFont.caption(SLCFontSize.bodySmall))
                                    .foregroundColor(SLCColor.textSecondary)
                            }
                            Spacer()
                            Image(systemName: "chevron.right")
                                .font(.system(size: 12))
                                .foregroundColor(SLCColor.textTertiary)
                        }
                    }
                }
                .buttonStyle(.plain)
                .padding(.horizontal, SLCSpace.md)
            }
        }
    }
}

// MARK: - 纪念日列表页（简单版）
struct AnniversaryListView: View {
    @EnvironmentObject var appState: AppState

    var body: some View {
        ScrollView {
            VStack(spacing: SLCSpace.sm) {
                ForEach(appState.anniversaries) { ann in
                    SLCCard {
                        HStack(spacing: SLCSpace.md) {
                            Text(ann.icon)
                                .font(.system(size: 36))
                            VStack(alignment: .leading, spacing: 4) {
                                Text(ann.title)
                                    .font(SLCFont.title(SLCFontSize.titleSmall, weight: .medium))
                                    .foregroundColor(SLCColor.textPrimary)
                                Text(DateUtils.fullChinese(ann.date))
                                    .font(SLCFont.caption(SLCFontSize.bodySmall))
                                    .foregroundColor(SLCColor.textSecondary)
                            }
                            Spacer()
                            VStack(alignment: .trailing) {
                                Text("还有 \(ann.daysUntilNext()) 天")
                                    .font(SLCFont.body(SLCFontSize.bodyMedium, weight: .medium))
                                    .foregroundColor(SLCColor.herDeep)
                            }
                        }
                    }
                    .padding(.horizontal, SLCSpace.md)
                }
            }
            .padding(.vertical, SLCSpace.md)
        }
        .background(SLCColor.cream)
        .navigationTitle("纪念日")
        .navigationBarTitleDisplayMode(.inline)
    }
}

#Preview {
    HomeView()
        .environmentObject(AppState.shared)
}

File: Android/app/src/main/java/com/slclovers/app/ui/home/HomeView.kt
package com.slclovers.app.ui.home

import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.slclovers.app.AppViewModel
import com.slclovers.app.data.model.AnniversaryEntity
import com.slclovers.app.data.model.TodoEntity
import com.slclovers.app.data.model.UserRole
import com.slclovers.app.ui.components.SLCCard
import com.slclovers.app.ui.components.SLCPersonBadge
import com.slclovers.app.ui.components.SLCSectionHeader
import com.slclovers.app.ui.theme.SLCColor
import com.slclovers.app.ui.theme.SLCRadius
import com.slclovers.app.ui.theme.SLCSpace
import com.slclovers.app.util.DateUtils
import kotlinx.coroutines.delay

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun HomeView(viewModel: AppViewModel) {
    val currentUser by viewModel.currentUser.collectAsState()
    val pairing by viewModel.pairing.collectAsState()
    val messages by viewModel.messages.collectAsState()
    val todos by viewModel.todos.collectAsState()
    val bucketItems by viewModel.bucketItems.collectAsState()
    val anniversaries by viewModel.anniversaries.collectAsState()

    // 每分钟刷新一次
    var now by remember { mutableStateOf(System.currentTimeMillis()) }
    LaunchedEffect(Unit) {
        while (true) {
            now = System.currentTimeMillis()
            delay(60_000)
        }
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(SLCColor.Cream)
            .verticalScroll(rememberScrollState())
            .padding(vertical = SLCSpace.md),
    ) {
        // 顶部
        HeaderSection(userName = currentUser?.name ?: "我")

        Spacer(Modifier.height(SLCSpace.lg))

        // 在一起大数字
        pairing?.startDate?.let { TogetherCounter(startDate = it, now = now) }

        Spacer(Modifier.height(SLCSpace.lg))

        // 每日寄语
        QuoteCard()

        Spacer(Modifier.height(SLCSpace.lg))

        // 下一个纪念日
        nextAnniversary(anniversaries)?.let { ann ->
            NextAnniversaryCard(anniversary = ann)
        }

        Spacer(Modifier.height(SLCSpace.lg))

        // 待办预览
        TodoPreviewSection(todos = todos.filter { !it.isCompleted }.take(3)) {
            todo -> viewModel.toggleTodo(todo)
        }

        Spacer(Modifier.height(SLCSpace.lg))

        // 愿望清单预览
        BucketPreviewSection(buckets = bucketItems.filter { !it.isAchieved }.take(5))

        Spacer(Modifier.height(SLCSpace.lg))

        // 最近聊天
        if (messages.isNotEmpty()) {
            RecentChatSection(lastMessage = messages.last(), partnerName = "TA")
        }

        Spacer(Modifier.height(SLCSpace.xxl))
    }
}

private fun nextAnniversary(anniversaries: List<AnniversaryEntity>): AnniversaryEntity? {
    return anniversaries
        .filter { !it.title.contains("在一起") }
        .minByOrNull { it.daysUntilNext() }
}

// ============ 顶部 ============
@Composable
private fun HeaderSection(userName: String) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = SLCSpace.lg),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Column {
            Text(
                greeting(),
                color = SLCColor.TextSecondary,
                fontSize = 13.sp,
            )
            Text(
                userName,
                fontWeight = FontWeight.SemiBold,
                fontSize = 28.sp,
                color = SLCColor.TextPrimary,
            )
        }
        Spacer(Modifier.weight(1f))
        Icon(
            Icons.Default.AccountCircle,
            contentDescription = "Profile",
            tint = SLCColor.Him,
            modifier = Modifier.size(36.dp)
        )
    }
}

private fun greeting(): String {
    val hour = java.util.Calendar.getInstance().get(java.util.Calendar.HOUR_OF_DAY)
    return when (hour) {
        in 5..10 -> "早安，"
        in 11..13 -> "中午好，"
        in 14..17 -> "下午好，"
        in 18..21 -> "晚上好，"
        else -> "夜深了，"
    }
}

// ============ 在一起大数字 ============
@Composable
private fun TogetherCounter(startDate: Long, now: Long) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = SLCSpace.lg)
            .shadow(2.dp, RoundedCornerShape(SLCRadius.xl))
            .clip(RoundedCornerShape(SLCRadius.xl))
            .background(
                Brush.verticalGradient(
                    listOf(SLCColor.CreamLight, SLCColor.Cream)
                )
            )
            .padding(vertical = SLCSpace.xl),
        contentAlignment = Alignment.Center
    ) {
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            Text(
                "在一起",
                color = SLCColor.TextSecondary,
                fontSize = 14.sp,
                letterSpacing = 4.sp,
            )
            Spacer(Modifier.height(SLCSpace.sm))

            Row(verticalAlignment = Alignment.Bottom) {
                Text(
                    DateUtils.togetherDays(startDate),
                    style = MaterialTheme.typography.displayLarge.copy(
                        fontSize = 96.sp,
                        fontWeight = FontWeight.UltraLight,
                    ),
                    color = SLCColor.Him,
                )
                Text(
                    " 天",
                    fontSize = 28.sp,
                    fontWeight = FontWeight.Light,
                    color = SLCColor.Him,
                    modifier = Modifier.padding(bottom = 12.dp)
                )
            }

            Spacer(Modifier.height(SLCSpace.sm))
            Text(
                DateUtils.fullChinese(startDate),
                color = SLCColor.TextSecondary,
                fontSize = 13.sp,
            )
        }
    }
}

// ============ 每日寄语 ============
@Composable
private fun QuoteCard() {
    val quotes = listOf(
        "和你在一起的每一天，都是最好的时光。",
        "遇见你，是所有故事里最美的那一页。",
        "愿我们携手，从心动走到古稀。",
        "最浪漫的事，是和你一起慢慢变老。",
        "世界再大，也不过是两个人的小家。",
        "你在哪里，哪里就是家。",
        "爱就是和心爱的人，一起做无聊的事。",
        "谢谢你，陪我走过每一个春夏秋冬。",
    )
    val today = java.util.Calendar.getInstance().apply {
        set(java.util.Calendar.HOUR_OF_DAY, 0)
        set(java.util.Calendar.MINUTE, 0)
        set(java.util.Calendar.SECOND, 0)
        set(java.util.Calendar.MILLISECOND, 0)
    }.timeInMillis
    val seed = (today / 86400000L).toInt()
    val quote = quotes[seed % quotes.size]

    Box(modifier = Modifier.padding(horizontal = SLCSpace.lg)) {
        SLCCard {
            Row {
                Icon(
                    Icons.Default.FormatQuote,
                    contentDescription = null,
                    tint = SLCColor.HerDeep.copy(alpha = 0.5f),
                    modifier = Modifier.size(24.dp)
                )
                Spacer(Modifier.width(SLCSpace.md))
                Text(
                    quote,
                    fontSize = 16.sp,
                    color = SLCColor.TextPrimary,
                    lineHeight = 22.sp,
                )
            }
        }
    }
}

// ============ 下一个纪念日 ============
@Composable
private fun NextAnniversaryCard(anniversary: AnniversaryEntity) {
    Box(modifier = Modifier.padding(horizontal = SLCSpace.lg)) {
        SLCCard {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(anniversary.icon, fontSize = 44.sp)
                Spacer(Modifier.width(SLCSpace.md))
                Column {
                    Text(
                        anniversary.title,
                        fontWeight = FontWeight.SemiBold,
                        fontSize = 18.sp,
                        color = SLCColor.TextPrimary,
                    )
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Text(
                            "还有 ",
                            fontSize = 13.sp,
                            color = SLCColor.TextSecondary,
                        )
                        Text(
                            "${anniversary.daysUntilNext()}",
                            style = MaterialTheme.typography.titleLarge.copy(
                                fontSize = 20.sp,
                                fontWeight = FontWeight.Medium,
                            ),
                            color = SLCColor.HerDeep,
                        )
                        Text(
                            " 天",
                            fontSize = 13.sp,
                            color = SLCColor.TextSecondary,
                        )
                    }
                }
                Spacer(Modifier.weight(1f))
                Icon(
                    Icons.Default.ChevronRight,
                    contentDescription = null,
                    tint = SLCColor.TextTertiary,
                )
            }
        }
    }
}

// ============ 待办预览 ============
@Composable
private fun TodoPreviewSection(todos: List<TodoEntity>, onToggle: (TodoEntity) -> Unit) {
    Column {
        SLCSectionHeader(title = "待办", action = "全部", onAction = { /* TODO */ })
        if (todos.isEmpty()) {
            Text(
                "暂无待办，去记下今天想做的事吧",
                color = SLCColor.TextSecondary,
                fontSize = 14.sp,
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(SLCSpace.lg)
                    .clip(RoundedCornerShape(SLCRadius.lg))
                    .background(SLCColor.CreamLight)
                    .padding(SLCSpace.lg),
                textAlign = androidx.compose.ui.text.style.TextAlign.Center,
            )
        } else {
            Column(
                modifier = Modifier.padding(horizontal = SLCSpace.md),
                verticalArrangement = Arrangement.spacedBy(SLCSpace.sm),
            ) {
                todos.forEach { todo ->
                    TodoRow(todo = todo, onToggle = { onToggle(todo) })
                }
            }
        }
    }
}

@Composable
private fun TodoRow(todo: TodoEntity, onToggle: () -> Unit) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(SLCRadius.md))
            .background(SLCColor.CreamLight)
            .padding(SLCSpace.md),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        IconButton(onClick = onToggle) {
            Icon(
                Icons.Default.RadioButtonUnchecked,
                contentDescription = "Toggle",
                tint = SLCColor.TextTertiary,
            )
        }
        Column(modifier = Modifier.weight(1f)) {
            Text(todo.title, fontSize = 16.sp, color = SLCColor.TextPrimary)
            todo.dueDate?.let {
                Text(
                    DateUtils.monthDay(it),
                    fontSize = 12.sp,
                    color = SLCColor.TextSecondary,
                )
            }
        }
        todo.assignee?.let { SLCPersonBadge(role = it, compact = true) }
    }
}

// ============ 愿望预览 ============
@Composable
private fun BucketPreviewSection(buckets: List<com.slclovers.app.data.model.BucketItemEntity>) {
    Column {
        SLCSectionHeader(title = "愿望清单")
        if (buckets.isEmpty()) {
            Text(
                "还没有愿望，去记下想做的事吧",
                color = SLCColor.TextSecondary,
                fontSize = 14.sp,
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(SLCSpace.lg)
                    .clip(RoundedCornerShape(SLCRadius.lg))
                    .background(SLCColor.CreamLight)
                    .padding(SLCSpace.lg),
                textAlign = androidx.compose.ui.text.style.TextAlign.Center,
            )
        } else {
            LazyRow(
                contentPadding = PaddingValues(horizontal = SLCSpace.md),
                horizontalArrangement = Arrangement.spacedBy(SLCSpace.sm),
            ) {
                items(buckets) { bucket ->
                    BucketCard(bucket)
                }
            }
        }
    }
}

@Composable
private fun BucketCard(bucket: com.slclovers.app.data.model.BucketItemEntity) {
    Column(
        modifier = Modifier
            .width(180.dp)
            .height(100.dp)
            .clip(RoundedCornerShape(SLCRadius.lg))
            .background(SLCColor.CreamLight)
            .padding(SLCSpace.md),
    ) {
        Text(
            bucket.title,
            fontWeight = FontWeight.Medium,
            fontSize = 16.sp,
            color = SLCColor.TextPrimary,
            maxLines = 2,
        )
        Spacer(Modifier.weight(1f))
        bucket.description?.let {
            Text(
                it,
                fontSize = 12.sp,
                color = SLCColor.TextSecondary,
                maxLines = 1,
            )
        }
    }
}

// ============ 最近聊天 ============
@Composable
private fun RecentChatSection(lastMessage: com.slclovers.app.data.model.MessageEntity, partnerName: String) {
    Column {
        SLCSectionHeader(title = "最近的消息")
        Box(modifier = Modifier.padding(horizontal = SLCSpace.md)) {
            SLCCard {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Icon(
                        Icons.Default.ChatBubble,
                        contentDescription = null,
                        tint = SLCColor.HerDeep,
                        modifier = Modifier.size(22.dp)
                    )
                    Spacer(Modifier.width(SLCSpace.md))
                    Column(modifier = Modifier.weight(1f)) {
                        Text(
                            lastMessage.content,
                            fontSize = 16.sp,
                            color = SLCColor.TextPrimary,
                            maxLines = 1,
                        )
                        Text(
                            DateUtils.friendlyRelative(lastMessage.createdAt),
                            fontSize = 12.sp,
                            color = SLCColor.TextSecondary,
                        )
                    }
                    Icon(
                        Icons.Default.ChevronRight,
                        contentDescription = null,
                        tint = SLCColor.TextTertiary,
                    )
                }
            }
        }
    }
}

// LazyRow helper - end of file